src/server.c

```diff
@@ -3416,23 +3423,22 @@ void call(client *c, int flags) {
      * c->cmd and c->lastcmd may be different, in case of MULTI-EXEC or
      * re-written commands such as EXPIRE, GEOADD, etc. */
 
-    /* Record the latency this command induced on the main thread.
-     * unless instructed by the caller not to log. (happens when processing
-     * a MULTI-EXEC from inside an AOF). */
-    if (flags & CMD_CALL_SLOWLOG) {
+    /* Log the command into the Slow log if needed and ecord the latency this command
+     * induced on the main thread unless instructed by the caller not to log.
+     * (happens when processing a MULTI-EXEC from inside an AOF).
+     * If the client is blocked we will handle slowlog when it is unblocked.
+     */
+    if ((flags & CMD_CALL_SLOWLOG) && !(c->flags & CLIENT_BLOCKED)) {
         char *latency_event = (real_cmd->flags & CMD_FAST) ?
                                "fast-command" : "command";
-        latencyAddSampleIfNeeded(latency_event,duration/1000);
+        latencyAddSampleIfNeeded(latency_event,c->duration/1000);
```

Comment:
So according to what you say a blocked command should be recorded twice? so for a "fast-command" which took longer we should get 2 events?

Reply:
I guess there is some kind of rational in this, since the monitor will let us understand WHEN the issue happened. It is just strange that we can potentially see more occurrences on the  latency monitor then the actual number of commands we processed

Reply:
done